app/com/gu/floodgate/Login.scala

```diff
@@ -4,26 +4,12 @@ import com.gu.googleauth._
 import play.api.libs.ws.WSClient
 import play.api.Configuration
 import com.gu.floodgate.views
+import play.api.mvc.Call
 import play.api.mvc.legacy.Controller
 
-trait AuthActions extends Actions with Filters {
-  def conf: Configuration
-  def groupChecker: GoogleGroupChecker = ???
-
-  // Google configuration
-  override val authConfig = GoogleAuthConfig(
-    clientId = conf.getOptional[String]("google.clientid").getOrElse(sys.error("No google clientid.")),
-    clientSecret = conf.getOptional[String]("google.clientsecret").getOrElse(sys.error("No google clientsecret.")),
-    redirectUrl = conf.getOptional[String]("google.oauthcallback").getOrElse(sys.error("No google oauthcallback.")),
-    domain = "guardian.co.uk"
-  )
-  // your app's routing
-  override val loginTarget = routes.Login.loginAction()
-  override val defaultRedirectTarget = routes.Application.index()
-  override val failureRedirectTarget = routes.Login.login()
-}
+import scala.concurrent.ExecutionContext
 
-class Login(val wsClient: WSClient, val conf: Configuration) extends Controller with AuthActions {
+class Login(val authConfig: GoogleAuthConfig, val wsClient: WSClient, val conf: Configuration)(implicit executionContext: ExecutionContext) extends Controller with LoginSupport {
```

Comment:
I don't know the `GoogleAuthConfig` workflow much, but shouldn't this `authConfig` object be used somewhere around, e.g. `startGoogleLogin`?